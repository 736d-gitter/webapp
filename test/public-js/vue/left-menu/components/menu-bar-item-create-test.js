'use strict';

const mount = require('../../vuex-mount');
const {
  default: MenuBarItemCreate
} = require('../../../../../public/js/vue/left-menu/components/menu-bar-item-create.vue');

describe('menu-bar-item-create', () => {
  it('create matches snapshot', () => {
    const { wrapper } = mount(MenuBarItemCreate, {
      type: 'create'
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('opens popover when item is clicked', () => {
    const { wrapper } = mount(MenuBarItemCreate, {
      type: 'create'
    });

    expect(wrapper.vm.popover).toEqual(undefined);

    wrapper.find({ ref: 'root' }).trigger('click');

    expect(wrapper.vm.popover).toBeDefined();
  });
});
