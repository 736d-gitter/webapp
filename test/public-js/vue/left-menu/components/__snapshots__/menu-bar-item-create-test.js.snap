// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu-bar-item-create create matches snapshot 1`] = `
<button
  class="item item-create"
  type="button"
>
  <span
    class="icon-wrapper"
  />
</button>
`;
